### ISSUE TITLE
Converting the Wiki over to GitIt and making portable packages for nixpkgs

### ISSUE THREAD

**Issue by third3ye** [NixOS/nixpkgs#3057]
Regarding: #1960 
Credit to: @chexxor, @deepfire

Since there has been a discussion about it (on the mailing list and here on github, though I'm to lazy to link) I thought I might as well setup an issue so that we don't flood the other non-specific issues with the particular subject of switching wiki solutions over to __GitIt__. We want to test out the __GitIt__ solution to let developers clone and commit to the wiki using their prefered markdown or markup languages, or alternatively use the __GitIt__ web interface, which I will help fit in to the new NixOS web-site web-design. If we'er succesfull in setting up an environment on the test box we have that everyone can agree on we will package the wiki data as _nixos-wiki_ and setup appropriate _NixOS_ services that can include _nixos-wiki_ as a build requirement. 

Here are the project specific goals:
* Replicate the _NixOS_ Wiki in it's entirety, save for the front page which will be remodeled and restructured by @chexxor
* Base the packaging and launching procedures on the focus of making __GitIt__ work with any (compatible) sshd and git backend available in _nixpkgs_ in a secure and stable manner.
* Create a process where a user can register using either <code>git-shell</code> through SSH or through the __GitIt__ web-interface and still be able to log in to either of the interfaces.
* For the sake of making the <code>nixos-wiki</code> package ready for multi-purpose use the contents of <code>nixos-wiki</code> should have a virtual GNU directory, like <code>/usr/doc/nixos/wikidata</code>, so that it fits in to a general UNiX environment that is not _NixOS_ and can be used by outside services without to much further a'do (debatable?)
* Setup the _NixOS_ services <code>services.gitit</code> (for GitIt standalone) and <code>services.gitit.nixos.org</code> (for fetching and setting up a standard _NixOS_ Wiki in to their <code>nix-env</code>). Both will have a set of options, which have not yet been decided upon.
* Incorporate the __GitIt__ templates with the <code>nixos-website</code> repo under "wiki" so that the template uses the libraries and assets of the main web-site to facilitate conformity.

You can check out the code for __GitIt__ here:
_https://github.com/jgm/gitit_
We've already setup a little server, you know the one:
__http://dreamsindigital.net__

It's all still very much an experiment, our attempt to bridge that developer-documentation gap. As it stands we're just trying to get __GitIt__ to run under expected conditions on the server we setup (__debian 7.0__ + __nix__), setup the best __GitIt__ conditions for the _Nix(OS) userbase_ and THEN finally making an environment and packaging it all for _NixOS_ in _nixpkgs_. All the tools are available in _nixpkgs_, so it's merely a question of pieceing it together in to a <code>nixos.services</code> expression, copy-pasting the data from the wiki or possibly getting that data on e-mail so we can be saved all that work and really put __Pandoc__ to the test, and then packaging the _nixos-wiki_ as data only (as it should really be a base for converting the wiki doc from one format to the other using pandoc, so that any package can include it in the markdown that they prefer, like if for instance you wanna make your own wiki-reading client in xmonad). 

Here's a lazy, copy-pasted half-assed attempt at making a registration process using git-shell, to start the conversation in one specific direction: how can the user register through either <code>git-shell</code> or the __GitIt__ web-interface and get access to both when registering in either one?

http://lpaste.net/106092

This is ment to be (at the moment) just a temporary fix to setup an automated process of getting users on to the GitIt wiki at dreamsindigital.net. As it stands the script is very hackish, faulty (in which way to point directories and files - if <code>git-shell</code> is restrictive enough - in the way that it's written in it's entirety) and shouldn't be used for anything else than currently testing this temporary setup and will eventually be replaced entirely with something that works in any _NixOS_ env setting and that makes backends shake to the same rhythm. But we would really appreciate some input on this, as making the user registration part is the largest hurdle in front of us right now (we are not l33t h4x0rs... we should be ashamed of our selves). Any suggestion in regards to making __GitIt__, _git_ and _sshd_ play nice and safely together would be appreciated.

I've tried very hard to meet the requirements and directives of _nix_. But any feedback on any misconceptions behind _nix_ or _NixOS_ that I might have missed is also very much appreciated.

In short: We have not setup pkgs yet and will not do so until we get some things clear.

Thanks for reading!

**Comment by michelk** [comment]
I also would appreciate, if this project would proceed. Maybe it  helps, if it moves to https://github.com/NixOS. @ocharles also has a section on nix on its [wiki](http://wiki.ocharles.org.uk/Nix), which he might merge.

**Comment by ttuegel** [comment]
This *needs* to proceed, one way or another. Aside from Mediawiki's annoying syntax and poor editing interface, it just *ate* some updates I tried to make to the wiki. That wouldn't happen with proper version control. If users can't reliably edit the wiki, then **we don't have a wiki**. </opinionated-rant>

**Comment by ocharles** [comment]
@michelk Actually, my plan there is to get that into the manual, as soon as `haskell-ng` is clear way to go.

**Comment by third3ye** [comment]
Again: this needs to be done. Not only does it mean the improved
portability of the Wiki data, but using the cross-markdown generator (I
forget it's name) we can easily adapt the wiki data to ANY new wiki
solution. Also; devs can rely on git rather than use the wiki pages, if
they want to... vis-a-vis: you give the user and developer the choice of
which method they want to access and edit the Wiki files.

**Comment by michelk** [comment]
How about using a similar approach as [haskelldocumentation](https://github.com/commercialhaskell/haskelldocumentation.git)?

**Comment by jagajaga** [comment]
So any continuation of the discussion?

**Comment by grahamc** [comment]
Looks like this can be closed in favor of the "Move the wiki" milestone.

**Comment by joachifm** [comment]
Agreed
